tpgtools/templates/resource.go.tmpl

```diff
@@ -235,9 +235,9 @@ func resource{{$.PathType}}Create(d *schema.ResourceData, meta interface{}) erro
 	}
 {{ end }}
 
-	id, err := {{ $.IdFunction }}(d, config, "{{$.ID}}")
```

Comment:
Does this mean we can remove `IdFunction` and `ID` now that the DCL exports an `ID` method?

Reply:
I believe so. Should I do that in this PR?

Reply:
Hmm... leave a TODO, maybe?  This does already include an awful lot of changes that are required-but-tangentially-related.

Reply:
Done

Reply:
I believe this would be better tracked if there were a GH issue that this TODO could point to